submitresult.go

```diff
@@ -35,6 +36,18 @@ func newResultClient(resultSubmitURL string, resultSubmitAPIToken stepconf.Secre
 }
 
 func (c *resultClient) uploadErrorResult(stepID string, err error) error {
+	if stepError, ok := err.(*step.Error); ok && len(stepError.Recommendations) > 0 {
```

Comment:
I would change the uploadErrorResult signature to include a reccomendation parameter.

Reply:
Discussed that it is minor issue.